### PR TITLE
Add test_image_processing_common.py

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Creates an equivalent `test_feature_extraction_common.py` for image processors: `test_image_processing_common.py` and moves any vision and image processing specific logic to this file. 

This is necessary for creating `ImageProcessingSavingTestMixin` in order to rename any feature extractor references in the image processor tests. This is left for a [future PR](https://github.com/huggingface/transformers/pull/20768). 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: tests/test_image_processing_common.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
+# coding=utf-8
+# Copyright 2021 HuggingFace Inc.
```
Comment (amyeroberts) [inline tests/test_image_processing_common.py:2]:
```suggestion
# Copyright 2023 HuggingFace Inc.
```

**Review by sgugger — APPROVED** [review]
Shouldn't all the tests use `ImageProcessingSavingTestMixin` instead of `FeatureExtractionSavingTestMixin`?

LGTM otherwise, thanks!

**Review by ydshieh — APPROVED** [review]
LGTM, with same question as Sylvain mentioned. Thank you for working on 🖼️ !

You mentioned

> This is necessary for creating ImageProcessingSavingTestMixin in order to rename any feature extractor references in the image processor tests. This is left for a https://github.com/huggingface/transformers/pull/20768.

But the new added `image_processing_common.py` contains `ImageProcessingSavingTestMixin`. So it looks a bit confusing, but I believe you have some reasons.

**Comment by amyeroberts** [comment]
@ydshieh @sgugger Yes, sorry, I could have been clearer above. 

The reason for not replacing `FeatureExtractionSavingTestMixin` with `ImageProcessingSavingTestMixin` in the `test_image_processing_xxx.py` files in this PR is that the tests in the original mixin use class attributes like `self.feature_extraction_class`. When updating the mixin, then attributes in the testing class `XxxImageProcessingTest` for each test file `test_image_processing_xxx.py` have to be updated. This results in either 1) updating just the FE references to have the tests running or 2) updating all of the FE references. In the case of 1) it results in the code being mixed between feature extractors and image processors which I found confusing to read (subjective opinion) and for 2) it introduced hundreds of lines of additional diff. I decided to leave the switch to a follow up PR.

**Comment by ydshieh** [comment]
> @ydshieh @sgugger Yes, sorry, I could have been clearer above.
> 
> The reason for not replacing `FeatureExtractionSavingTestMixin` with `ImageProcessingSavingTestMixin` in the `test_image_processing_xxx.py` files in this PR is that the tests in the original mixin use class attributes like `self.feature_extraction_class`. When updating the mixin, then attributes in the testing class `XxxImageProcessingTest` for each test file `test_image_processing_xxx.py` have to be updated. This results in either 1) updating just the FE references to have the tests running or 2) updating all of the FE references. In the case of 1) it results in the code being mixed between feature extractors and image processors which I found confusing to read (subjective opinion) and for 2) it introduced hundreds of lines of additional diff. I decided to leave the switch to a follow up PR.

Understand! Thank you for explaining!
BTW, it would be nice to give a link like https://github.com/huggingface/transformers/blob/997cbebf3483d500de8f85cc8834b704f6b410be/tests/models/beit/test_image_processing_beit.py#L110
(I agree it is a bit more consuming :-) )

**Pull request merged**